### PR TITLE
Switch to  gicv3

### PR DESCRIPTION
**Pull request by dsemenets**

### CODE REVIEW

File: boards/arm64/xenvm/xenvm-qemu.dts [hunk 1]
```diff
@@ -28,6 +28,7 @@
 		zephyr,sram = &ram;
 		zephyr,console = &xen_consoleio_hvc;
```
Comment (oleksiimoisieiev) [inline boards/arm64/xenvm/xenvm-qemu.dts:29]: Why Flash SIMULATOR is in this MR?

File: boards/arm64/xenvm/xenvm-qemu.dts [hunk 2]
```diff
@@ -82,4 +83,45 @@
 		compatible = "xen,uart_hvc";
 		status = "okay";
 	};
+	flashcontroller0: flashcontroller {
+		compatible = "zephyr,sim-flash";
```
Comment (oleksiimoisieiev) [inline boards/arm64/xenvm/xenvm-qemu.dts:88]: I think we still need to be able to select which gic do we need.

File: boards/arm64/xenvm/xenvm_defconfig [hunk 3]
```diff
@@ -33,9 +34,23 @@ CONFIG_MAIN_STACK_SIZE=32768
...
 CONFIG_ISR_STACK_SIZE=32768
 CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=32768
-CONFIG_CMSIS_THREAD_MAX_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_MAX_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE=32768
-CONFIG_IPM_CONSOLE_STACK_SIZE=32768
+#CONFIG_CMSIS_THREAD_MAX_STACK_SIZE=32768
+#CONFIG_CMSIS_V2_THREAD_MAX_STACK_SIZE=32768
+#CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE=32768
+#CONFIG_IPM_CONSOLE_STACK_SIZE=32768
```
Comment (oleksiimoisieiev) [inline boards/arm64/xenvm/xenvm_defconfig:40]: Please do not push commented code. It's not obvious what are these changes for.

File: soc/arm64/xenvm/soc.h [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#ifndef _SOC_H
+#define _SOC_H
```
Comment (oleksiimoisieiev) [inline soc/arm64/xenvm/soc.h:2]: Why do we need this empty header?

**Comment by oleksiimoisieiev** [comment]
Also please update commit messages. Make them complete, so your commit can be cherry-picked to features.

File: boards/arm64/xenvm/board.cmake [hunk 5]
```diff
@@ -3,13 +3,14 @@ set(QEMU_ARCH aarch64)
 
 set(QEMU_CPU_TYPE_${ARCH} cortex-a57)
 
+set(QEMU_GIC_VERSION 3)
 set(QEMU_KERNEL_OPTION -kernel ${ZEPHYR_BINARY_DIR}/xen)
 set(QEMU_FLAGS_${ARCH}
    -gdb tcp::4224
    -cpu ${QEMU_CPU_TYPE_${ARCH}} -m 6G
    -nographic
```
Comment (oleksiimoisieiev) [inline boards/arm64/xenvm/board.cmake:11]: I think we still need to make GIC verision optional. I see this implementation as to create separate configs to handle both gicv2 and gicv3

**Comment by oleksiimoisieiev** [comment]
I think there should be some parameter, exposed to the code so correct gic version can be placed in the domain_config.

**Comment by oleksiimoisieiev** [comment]
I suggest renaming CONFIG_XENVM_USE_GIC_V3 to CONFIG_XEN_USE_GIC_V3 so this config can be used for the future configs. Also I think CONFIG_XEN_USE_GIC_V3 should be mentioned in KConfig with appropriate dependencies.

**Comment by dsemenets** [comment]
> I suggest renaming CONFIG_XENVM_USE_GIC_V3 to CONFIG_XEN_USE_GIC_V3
I don't think it is reasonable. This option created for xenvm board not for xen subsystem

File: boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig [hunk 6]
```diff
@@ -0,0 +1,31 @@
+CONFIG_SOC_XENVM=y
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig:1]: Do we actually need whole new defconfig for just 1 changed option? I think it is better to left default GICv3 for xenvm_qemu and, if someone will need GICv2 in qemu, just change this value on own setup.

**Review by firscity — COMMENTED** [review]
Please, no short or empty commit messages. Such commits will not be merged.

File: boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig (hunk 6 above)
Comment (dsemenets) [reply boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig:1]: There is more then 1 option.
There is different reg value in dts. I wanted to do this switcher by option but I can't use different dts for this. Some reason dts processed before any of CMakeList.txt I can change (in boards/arm64/xenvm/ and application directories)

File: boards/arm64/xenvm/board.cmake [hunk 7]
```diff
@@ -2,14 +2,18 @@ set(SUPPORTED_EMU_PLATFORMS qemu)
...
+else()
+    set(QEMU_GIC_VERSION 2)
+endif()
 set(QEMU_KERNEL_OPTION -kernel ${ZEPHYR_BINARY_DIR}/xen)
 set(QEMU_FLAGS_${ARCH}
    -gdb tcp::4224
    -cpu ${QEMU_CPU_TYPE_${ARCH}} -m 6G
    -nographic
-   -machine virt,gic-version=2,virtualization=true
+   -machine virt,gic-version=${QEMU_GIC_VERSION},virtualization=true
```
Comment (firscity) [inline boards/arm64/xenvm/board.cmake:15]: Are this changes needed if we decided to build/run qemu separately from Zephyr (build with Moulin/run with separate script)? I don`t think that Zephyr OS kernel is proper place for this. Also this will be broken with zephyr-dom0-xt, because it doesn't build Xen.

File: boards/arm64/xenvm/xenvm-qemu.dts [hunk 8]
```diff
@@ -11,75 +11,9 @@
  * with zephyr requirements.
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-qemu.dts:11]: This description is not correct anymore. Also copyright should be `2023`, it was not created in 2020

File: boards/arm64/xenvm/xenvm-gicv2-qemu.dts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2020 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This file was created by running
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-gicv2-qemu.dts:6]: This is not true, please, remove this comment and set copyright to 2023.

File: boards/arm64/xenvm/xenvm-qemu.dtsi [hunk 10]
```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2020 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This file was created by running
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-qemu.dtsi:6]: Copyright and comment again.

File: boards/arm64/xenvm/board.cmake (hunk 7 above)
Comment (dsemenets) [reply boards/arm64/xenvm/board.cmake:15]: I did this before we decided that. Other hand I had to make test. As far as I know we didn't move to run qemu separately from Zephyr

File: boards/arm64/xenvm/xenvm-qemu.dtsi [hunk 11]
```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This file was created by running
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-qemu.dtsi:6]: This file was NOT created by LIBXL_DUMP_DTB, it is manually created for qemu Dom0. Please, remove or edit this comment.

File: boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+CONFIG_AARCH64_IMAGE_HEADER=y
+CONFIG_MAX_XLAT_TABLES=10
+CONFIG_HEAP_MEM_POOL_SIZE=262144
+
+# Enable console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# Enable PV hypervisor console
+#CONFIG_UART_CONSOLEIO_XEN_HVC=y
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig:20]: Please fix this, no commented configs, this issue is already fixed in this branch. Please, check the correct name for this config.

File: boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+# Enable logging subsys
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=n
+
+CONFIG_TEST=y
+#CONFIG_PRINTK_SYNC=y
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig:27]: No commented configs.

**Review by firscity — CHANGES_REQUESTED** [review]
Also, no empty or short commit messages are allowed.

File: boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig (hunk 13 above)
Comment (dsemenets) [reply boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig]: Done

File: boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig (hunk 12 above)
Comment (dsemenets) [reply boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig]: Done

File: boards/arm64/xenvm/xenvm-qemu.dtsi (hunk 11 above)
Comment (dsemenets) [reply boards/arm64/xenvm/xenvm-qemu.dtsi]: Done

File: boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+CONFIG_AARCH64_IMAGE_HEADER=y
+CONFIG_MAX_XLAT_TABLES=10
+CONFIG_HEAP_MEM_POOL_SIZE=262144
+
+# Enable console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# Enable PV hypervisor console
+CONFIG_UART_CONSOLEIO_XEN_HVC=y
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig:20]: This code is not buildable.` CONFIG_UART_CONSOLEIO_XEN_HVC `-> `CONFIG_UART_XEN_HVC_CONSOLEIO`

File: boards/arm64/xenvm/xenvm-qemu_defconfig [hunk 15]
```diff
@@ -26,7 +26,6 @@ CONFIG_LOG=y
 CONFIG_LOG_MODE_MINIMAL=n
 
 CONFIG_TEST=y
-#CONFIG_PRINTK_SYNC=y
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-qemu_defconfig]: Looks like spurious change. It is right, but at least it should be in separate commit.

**Comment by firscity** [comment]
@dsemenets, please add more appropriate commit message for commit 1 and 2. It should contain about motivation and point of the changes, not about copying commit topic (at least 1 paragraph of text, as was described by @lorc some time ago).

Commit 3 will be dropped from git history, so it may be unchanged.

File: boards/arm64/xenvm/xenvm-gicv3-qemu_defconfig [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+
+# Enable PV hypervisor console
+CONFIG_UART_XEN_HVC_CONSOLEIO=y
+
+# Enable logging subsys
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=n
+
+CONFIG_TEST=y
+#CONFIG_PRINTK_SYNC=y
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-gicv3-qemu_defconfig]: Please remove commented config

File: boards/arm64/xenvm/xenvm-qemu.dts [hunk 17]
```diff
@@ -1,85 +1,13 @@
...
-		label = "arch_timer";
-	};
-
-	hypervisor: hypervisor@48200000 {
-		compatible = "xen,xen-4.15", "xen,xen";
-		reg = <0x00 0x48200000 0x00 0x1000000>;
-		interrupts = <GIC_PPI 0x0 IRQ_TYPE_EDGE IRQ_DEFAULT_PRIORITY>;
-		interrupt-parent = <&gic>;
-	};
+#include "xenvm-qemu.dtsi"
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-qemu.dts:9]: Redundant whitespace

**Comment by oleksiimoisieiev** [comment]
Why did you add gicv2/3 switch to xenvm-qemu only? We need this to build xenvm as well. Also I think this switch can be done as overlay for both boards.

**Comment by dsemenets** [comment]
We already have xenvm_gicv3 board for gicv3

**Review by firscity — APPROVED** [review]
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`
